### PR TITLE
[FLYD-55] Enhance submissions form page

### PR DESCRIPTION
**Pull request by nkoul**
Adds a better editor for users to submit content. Automatically sets the creator of the submission to the person who is logged in. View testing added as well.

### CODE REVIEW

File: app/views/categories/show.html.erb [hunk 1]
```diff
@@ -8,8 +8,16 @@
             <div class="index__submission">
                 <div class="index__submission-vote-group">
                     <div class="index__vote">
-                        <% vote = Vote.find_by(post: submission, user_id: current_user.id) %>
-						<% if vote.nil? || vote. weight == 0 %> 
+                        <% vote = Vote.find_by(post_id: submission.id, user_id: current_user.id) %>
+                        <% if vote.nil? || vote. weight == 0 %>
```
Comment (clburns107) [inline app/views/categories/show.html.erb]: should it be ```vote.weight``` (without the extra space)?

File: app/views/submissions/show.html.erb [hunk 2]
```diff
@@ -1,31 +1,59 @@
...
-<%= render partial: "new_comment", locals: {parent_comment: nil} %>
-<% comments = @submission.comments.order(vote_count: :desc, created_at: :desc).where(parent_comment: nil) %>
-<%= render partial: "comments", locals: {comments: comments, parent_comment: nil} %> 
-<div>
+<div class = "show index">
+	<div class = "show__dropdownVoteGroup">
+		<div class = "show__voteTitleGroup">
+			<div class="vote">
+				<% vote = Vote.find_by(post_id: @submission.id, user_id: current_user.id) %>
+				<% if vote.nil? || vote. weight == 0 %>
```
Comment (clburns107) [inline app/views/submissions/show.html.erb]: ```vote.weight```

File: spec/views/submissions/new.html.erb_spec.rb [hunk 3]
```diff
@@ -3,6 +3,9 @@
 RSpec.describe "submissions/new", type: :view do
   before(:each) do
     assign(:submission, Submission.new())
+    @user = create(:user)
+    #login(@user)
```
Comment (bethanyhaubert) [inline spec/views/submissions/new.html.erb_spec.rb]: can you get rid of this line you commented out?

File: spec/models/vote_spec.rb [hunk 4]
```diff
@@ -17,11 +21,7 @@
 		expect(vote).to be_valid 	
 	end 
 
-	it "denies duplicate creation" do
-		vote1 = Vote.create(post: @sub, user: @user)
-		vote2 = Vote.new(post: @sub, user: @user)
-		expect(vote2).to be_invalid
-	end
+
```
Comment (bethanyhaubert) [inline spec/models/vote_spec.rb]: extra line  you can get rid of

**Pull request merged**